### PR TITLE
[Java] Remove unnecessary allocation during String field emptiness checks

### PR DESCRIPTION
**Pull request by hpple**
Hi!

I've opened an issue about suspicious String field processing earlier:
#6719 

This patch is trying to fix code generation in case this is actually a bug.

### CODE REVIEW

**Comment by hpple** [comment]
@googlebot I signed it!

**Comment by hpple** [comment]
Can you take a look please? The change list is relatively small.

**Comment by hpple** [comment]
Any chance to get a code review?

File: src/google/protobuf/compiler/java/java_string_field.cc [hunk 1]
```diff
@@ -110,7 +112,7 @@ void SetPrimitiveVariables(const FieldDescriptor* descriptor,
     (*variables)["clear_has_field_bit_builder"] = "";
 
     (*variables)["is_field_present_message"] =
-        "!get" + (*variables)["capitalized_name"] + "Bytes().isEmpty()";
+       "!" + (*variables)["isStringEmpty"] + "(" + (*variables)["name"] + "_)";
```
Comment (fowles) [inline src/google/protobuf/compiler/java/java_string_field.cc:115]: Don't we know the type in the generated code such that we could call `(*variables)["name"] + "_.isEmpty()"` here directly?

Comment (hpple) [reply src/google/protobuf/compiler/java/java_string_field.cc:115]: Unfortunately, no.

As I initially described it in https://github.com/protocolbuffers/protobuf/issues/6719, the problem is that (for some reason) fields for **string** type in proto are generated as **java.lang.Object** in Java. So, such fields may contain either **ByteString** or **java.util.String** depending on which representation was requested last time.

**Pull request merged**